### PR TITLE
LO-5874: Load Data Lake Results to Dataframe

### PR DESCRIPTION
**Pull request by mjtieman**
These changes add methods to f load data lake query results to a Pandas dataframe. Methods to fetch the data lake table schemas have also been added.

- `execute_data_lake_query_to_dataframe` Executes a data lake query, downloads the result file and converts to a Pandas dataframe
- `load_data_lake_result_to_dataframe` Downloads the result file of a query and converts to a Pandas dataframe
- `list_data_lake_schemas` Fetches the data lake table schemas
- `get_data_lake_schema` Fetches the schema for a specific data lake table

A method to check if a specific file exists was also added to the files service.

### CODE REVIEW

File: phc/services/analytics.py [hunk 1]
```diff
@@ -133,3 +144,261 @@ def get_data_lake_query(self, query_id: str) -> ApiResponse:
             The data lake query get response
         """
         return self._api_call("analytics/query/%s" % query_id, http_verb="GET")
+
+    def list_data_lake_schemas(self, dataset_id: str) -> ApiResponse:
```
Comment (mschroering) [inline phc/services/analytics.py]: We've been trying to use `project_id` as the public facing sdk name across the different resources.

Comment (mjtieman) [reply phc/services/analytics.py]: Done

File: phc/services/analytics.py [hunk 2]
```diff
@@ -133,3 +144,261 @@ def get_data_lake_query(self, query_id: str) -> ApiResponse:
...
+
+        Returns
+        -------
+        phc.ApiResponse
+            The schema for each data lake table
+        """
+        path = "analytics/data-lake/schema?datasetId=%s" % (project_id)
+        return self._api_call(path, http_verb="GET")
+
+    def get_data_lake_schema(self, project_id, table) -> ApiResponse:
```
Comment (mschroering) [inline phc/services/analytics.py]: Missing type hints.

File: phc/services/analytics.py [hunk 3]
```diff
@@ -133,3 +144,261 @@ def get_data_lake_query(self, query_id: str) -> ApiResponse:
...
+        )
+        if not self.__poll_predicate(files_client.exists, 10, query_id):
+            raise RuntimeError(
+                f"Timed out waiting for result file {query_id} to become available"
+            )
+
+        download_path = files_client.download(query_id)
+        return _pd.read_csv(download_path)
+
+    def __data_lake_query_predicate(self, analytics_client, query_id):
```
Comment (mschroering) [inline phc/services/analytics.py]: Missing type hints

File: phc/services/analytics.py (hunk 2 above)
Comment (mjtieman) [reply phc/services/analytics.py]: Done

**Pull request merged**